File: hackerearth/3SUM.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int n = nums.size();
        set<vector<int>>curr;
        int target=0;
        vector<vector<int>>result;
        for(int i=0;i<n;i++){
        int j=i+1;
        int k=n-1;
        while(j<k){
            int sum=nums[i]+nums[j]+nums[k];
            if(sum==target){
                curr.insert({nums[i],nums[j],nums[k]});
                j++;
                k--;
            }
            else if(sum<target){
                j++;
            }
            else{
                k--;
            }
        }
        }
        for(auto a:curr){
            result.push_back(a);
        }
        return result;
    }
};

File: leetcode/3SUM.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        n = len(nums)
        ans=set()
        for i in range(n-2):
            second = i+1
            third = n-1
            while(second<third):
                sum = nums[i]+nums[second]+nums[third]
                if(sum<0):
                    second+=1
                elif(sum>0):
                    third-=1
                else:
                    ans.add((nums[i],nums[second],nums[third]))
                    second+=1
                    third-=1
        return ans
